packages/trip-form/src/MetroModeSelector/SubSettingsPane.tsx

```diff
@@ -178,6 +189,34 @@ export default function SubSettingsPane({
     { settingsNoSubmodes: [], settingsOnlySubmodes: [] }
   );
 
+  const handleSettingChange = useCallback(
+    (setting: ModeSetting) => (evt: QueryParamChangeEvent) => {
+      // rental mode settings do not have type "SUBMODE"
+      const settingsWithTransportMode = modeButton.modeSettings.filter(
+        (s: ModeSetting) =>
+          (s.type === "CHECKBOX" || s.type === "SUBMODE") && s.addTransportMode
+      );
+      // check if setting is a transport mode setting
```

Comment:
You can probably move this variable outside of the `useCallback` block, e.g. move to line 191.